pkg/yurtappmanager/controller/uniteddeployment/uniteddeployment_controller_utils.go

```diff
@@ -97,3 +102,18 @@ func filterOutCondition(conditions []unitv1alpha1.UnitedDeploymentCondition, con
 	}
 	return newConditions
 }
+
+func GetNextPatches(ud *unitv1alpha1.UnitedDeployment) map[string]UnitedDeploymentPatches{
+	next := make(map[string]UnitedDeploymentPatches)
+	for _, pool := range ud.Spec.Topology.Pools {
+		t := UnitedDeploymentPatches{}
+		if pool.Replicas != nil {
+			t.Replicas = *pool.Replicas
+		}
+		if pool.Patches != nil {
```

Comment:
You may need to think about a problem here. 

If the Patch changes the "replica", the replica value in the UnitedDeploymentPatches does not match the final replica number.

Reply:
`Patch` takes precedence over `Replicas` fields. If the `Patch` also modifies the replicas, use the replicas value in the Patch.